5.5/test/test_replication.sh

```diff
@@ -0,0 +1,169 @@
+  docker logs openshift-origin
+}
+
+function dump_pod_logs() {
+  echo "Dumping MySQL master logs:"
+  run "oc logs $(get_pod_name mysql-master)"
+  echo "Dumping MySQL slave logs:"
+  run "oc logs $(get_pod_name mysql-slave)"
+}
+
+#
+# Helper functions to run commands inside of the OpenShift Docker container.
+#
+function run() {
+  docker exec openshift-origin /bin/bash -c "$@"
+}
+
+function run_interactive() {
+  docker exec -i openshift-origin /bin/bash -c "$@"
+}
+
+function get_pod_name() {
+  name=$1; shift
+  run "oc get pods | grep $name | cut -f 1 -d ' '" | head -n 1
+}
+
+
+# Used for substituting the image name in the template.
+function substitute_image() {
+  sed -e "s|openshift/mysql-55-centos7|${IMAGE_NAME:-openshift/mysql-55-centos7}|"
+}
+
+# Main
+setup_dns
+start_openshift
+# FIXME: Make sure our testing vagrant box has this package installed
+yum -y install nfs-utils
+set +x
+echo "Waiting for OpenShift to start ..."
+wait_for_url_timed "https://${HOST_DOCKER_IP}:8443/healthz" "" 90*TIME_SEC >/dev/null
+set -x
+
+# Install mysql command line tool so we can make test queries.
+run "yum -y install mysql"
+
+# Create new project.
+run "oc new-project replication"
+
+# Allow all pods to run as privileged, so NFS server works.
+# FIXME: There doesn't seem to be a nice way to simply add a new user to an SCC
+run "oc patch scc privileged -p '{\"users\":[\"system:serviceaccount:openshift-infra:build-controller\",\"system:serviceaccount:replication:default\"]}'"
+
+# Create Persistent Volumes backed by NFS.
+cat examples/replica/nfs-pv-provider.json | run_interactive "oc process -f - | oc create -f -"
+
+# Now create MySQL replica scenario.
+cat examples/replica/mysql_replica.json | substitute_image | run_interactive "oc process -f - | oc create -f -"
+
+# Wait until master and slave are up.
+set +x
+echo "Waiting for MySQL Master and Replica to come online ..."
+trap dump_openshift_logs ERR
+wait_for_url_timed "mysql-master.replication.svc.cluster.local:3306" "" 5*TIME_MIN >/dev/null
+wait_for_url_timed "mysql-slave.replication.svc.cluster.local:3306" "" 1*TIME_MIN >/dev/null
+trap - ERR
+set -x
+
+# Get environment variables from master, so we know the passwords.
+export $(run "oc env --list dc mysql-master | grep -v '^#'")
+
+#
+# Tests start here.
+#
+
+trap dump_pod_logs ERR
+
+# Create table in master.
+run_interactive "mysql -h mysql-master.replication.svc.cluster.local -u${MYSQL_USER} -p${MYSQL_PASSWORD} ${MYSQL_DATABASE}" <<EOF
+CREATE TABLE tbl (col1 VARCHAR(20), col2 VARCHAR(20));
+INSERT INTO tbl VALUES ('foo1', 'bar1');
+EOF
+
+sleep 2
+
+# Verify that values are replicated in slave.
+SLAVE_POD_NAME=$(get_pod_name mysql-slave)
+# FIXME: Should log in as normal user, doesn't work right now.
+run "oc exec $SLAVE_POD_NAME -- /bin/bash -c 'mysql -uroot ${MYSQL_DATABASE} -e \"SELECT * FROM tbl;\"'" | grep -q foo1
+
+echo "All tests finished successfully, cleaning up"
+run "oc delete all --all"
```

Comment:
`oc delete all --all` works, but operates on a single project/namespace, need to run it for every namespace (including 'default') -- that will stop containers, but not stop OS.